### DISCUSSION TITLE
Cannot use server actions in same file?

### DISCUSSION THREAD

**Discussion by vedantroy** [vercel/next.js#50543] [Help] [answered]
### Summary

Not if this is a bug or not (in which case I'll open an issue), but I can't seem to use server actions in the same file. I need to import them from a different file.

### Additional information

```js
Code:

"use server";

export default async function() {
  async function createGame(formData) {
    "use server";
    const obj = Object.fromEntries(formData);
    const gameId = obj["game"];
    console.log(gameId);
  }

  return (
    <>
        {[{id: 1}, {id: 2}, {id: 3}].map((item) => (
        <form action={createGame}>
          <button
            name="game"
            value={item.id}
            type="submit"
            key={item.id}
            className="p-4 hover:bg-gray-100 rounded-lg cursor-pointer"
          >
          </button>
        </form>
      ))}
    </>
  );
}
```

### Example

https://github.com/vedantroy/nextjs-bug-reproduction-2

**Comment by icyJoseph** [comment]
Why do you have a top level `"use server"` directive here? All components are Server Components unless marked as `"use client"` or if used within a `"use client"` directive tree. 

The `use server` directive does nothing to components, AFAIK. If you want to poison this module to be server side only, you need to install `server-only` and use that: https://nextjs.org/docs/getting-started/react-essentials#the-server-only-package

**Comment by vedantroy** [comment, marked as answer]
Removing "use server" solved the problem--this is unintuitive though.
